### PR TITLE
fix(mrgn-ui): add fallbacks to failing oracle prices from crossbar

### PR DESCRIPTION
**Pull request by borcherd**
this includes:
- fallbacks when oracle price fails
- NaN catching
- stake page fixes

### CODE REVIEW

File: apps/marginfi-v2-trading/src/pages/api/oracle/price.ts [hunk 1]
```diff
@@ -154,16 +200,115 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
...
+      return {
+        feedHash,
+        results: [price.value],
+      };
+    });
+
+    return { payload: finalPayload, brokenFeeds };
+  } catch (error) {
+    console.error("Error:", error);
+    return { payload: [], brokenFeeds: feedHashes };
```
Comment (borcherd) [inline apps/marginfi-v2-trading/src/pages/api/oracle/price.ts:288]: You're returning all feedHashes here as broken, should we do that?

File: apps/marginfi-v2-trading/src/pages/api/oracle/price.ts [hunk 2]
```diff
@@ -154,16 +200,115 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
...
+  const isAuth = username && bearer;
+
+  const basicAuth = isAuth ? Buffer.from(`${username}:${bearer}`).toString("base64") : undefined;
 
   try {
     const feedHashesString = feedHashes.join(",");
-    const response = await fetch(`${SWITCHBOARD_CROSSSBAR_API}/simulate/${feedHashesString}`, {
+    const response = await fetch(`${endpoint}/simulate/${feedHashesString}`, {
       headers: {
+        Authorization: basicAuth ? `Basic ${basicAuth}` : "",
```
Comment (borcherd) [inline apps/marginfi-v2-trading/src/pages/api/oracle/price.ts:311]: Basic ${basicAuth} -> should this have a space inbetween? Also, doesnt it cause any issues if we have an empty Authorization string?

File: apps/marginfi-v2-trading/src/pages/api/oracle/price.ts [hunk 3]
```diff
@@ -241,3 +389,50 @@ function extractHost(referer: string | undefined): string | undefined {
...
+    [tokenAddress: string]: BirdeyeTokenPrice;
+  };
+}
+
+async function fetchMultiPrice(tokens: string[]): Promise<BirdeyePriceResponse> {
+  if (!tokens) {
+    throw new Error("No tokens provided");
+  }
+
+  // use abort controller to restrict fetch to 10 seconds
```
Comment (borcherd) [inline apps/marginfi-v2-trading/src/pages/api/oracle/price.ts:414]: You say 10 secs but is actually 5

File: apps/marginfi-v2-ui/src/pages/api/lst.ts [hunk 4]
```diff
@@ -19,6 +19,10 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse<
     apy: 0,
   };
 
+  if (!process.env.VALIDATOR_API_URL) {
```
Comment (borcherd) [inline apps/marginfi-v2-ui/src/pages/api/lst.ts:22]: nicu

File: apps/marginfi-v2-ui/src/pages/api/oracle/price.ts [hunk 5]
```diff
@@ -18,6 +18,7 @@ import {
   FeedResponse,
 } from "@mrgnlabs/marginfi-client-v2/dist/vendor";
 import { chunkedGetRawMultipleAccountInfoOrdered, median, Wallet } from "@mrgnlabs/mrgn-common";
+import { push } from "@socialgouv/matomo-next";
```
Comment (borcherd) [inline apps/marginfi-v2-ui/src/pages/api/oracle/price.ts:21]: whats this?

File: apps/marginfi-v2-ui/src/pages/api/oracle/price.ts [hunk 6]
```diff
@@ -102,17 +111,38 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
     for (const index in requestedOraclesData) {
       const oracleData = requestedOraclesData[index];
       const priceDataRaw = oracleAis[index];
-      const oraclePrice = parsePriceInfo(oracleData.oracleSetup, priceDataRaw.data);
+      const mintData = oracleMintMap.get(oracleData.oracleKey)!;
```
Comment (borcherd) [inline apps/marginfi-v2-ui/src/pages/api/oracle/price.ts:114]: Isnt it dangerous to add ! at the end?

File: apps/marginfi-v2-trading/src/pages/api/oracle/price.ts (hunk 1 above)
Comment (k0beLeenders) [reply apps/marginfi-v2-trading/src/pages/api/oracle/price.ts:288]: yeah if the api call fails that means, non of the feeds will be resolved thus moving those feeds to the next fallback

File: apps/marginfi-v2-trading/src/pages/api/oracle/price.ts (hunk 2 above)
Comment (k0beLeenders) [reply apps/marginfi-v2-trading/src/pages/api/oracle/price.ts:311]: yeah the space is needed, the public crossbar api doesn't need authorization so this header is ignored

File: apps/marginfi-v2-ui/src/pages/api/oracle/price.ts (hunk 6 above)
Comment (k0beLeenders) [reply apps/marginfi-v2-ui/src/pages/api/oracle/price.ts:114]: it should never be undefined, feel free to correct me if I'm wrong

File: apps/marginfi-v2-ui/src/pages/api/oracle/price.ts (hunk 5 above)
Comment (k0beLeenders) [reply apps/marginfi-v2-ui/src/pages/api/oracle/price.ts:21]: should be removed, good catch

**Pull request merged**